src/ast.hpp

```diff
@@ -218,7 +208,7 @@ namespace Sass {
   // "Template Method" design pattern to allow subclasses to adjust their flags
   // when certain objects are pushed.
   /////////////////////////////////////////////////////////////////////////////
-  template <typename T>
+  template <typename T, typename U>
```

Comment:
This `U` for chaining seems like unnecessary complexity for little/no gain. Let's not do this, the `Vectorized` class is already quite convoluted.

Reply:
I propose to do this later. The whole Vectorized and Hashed needs overhauling.
And I don't think it adds that much of complexity ...

Reply:
> I propose to do this later.

Why do it here, if it's not even being used (as far as I can tell by searching through the diffs)?

Reply:
https://github.com/sass/libsass/pull/2908/files/5ed7e8429c1138de625f3fa40a340fcabedaca2a#diff-dd77ff8326ef81051bd3b8cc5cc33669

Reply:
Since I'm tired of debating it, I removed the Vector method chaining experiment.